test/lit/passes/optimize-instructions-bulk-memory.wast

```diff
@@ -10,10 +10,17 @@
   ;; CHECK-NEXT:   (local.get $dst)
   ;; CHECK-NEXT:   (local.get $sz)
   ;; CHECK-NEXT:  )
-  ;; CHECK-NEXT:  (memory.copy
-  ;; CHECK-NEXT:   (local.get $dst)
-  ;; CHECK-NEXT:   (local.get $src)
-  ;; CHECK-NEXT:   (i32.const 0)
+  ;; CHECK-NEXT:  (block
+  ;; CHECK-NEXT:   (drop
+  ;; CHECK-NEXT:    (i32.load8_u
+  ;; CHECK-NEXT:     (local.get $dst)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:   (drop
+  ;; CHECK-NEXT:    (i32.load8_u
+  ;; CHECK-NEXT:     (local.get $src)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
```

Comment:
I guess we could move further and check `$dst` or `$src` for constant address and if `memory.init >= (C >> 16)`  eliminate even `load`